back/scripts/datasets/dataset_aggregator.py

```diff
@@ -0,0 +1,150 @@
+import hashlib
+import json
+import logging
+import urllib
+from collections import defaultdict
+from pathlib import Path
+from urllib.error import HTTPError
+
+import pandas as pd
+import polars as pl
+from tqdm import tqdm
+
+from back.scripts.loaders import LOADER_CLASSES
+from back.scripts.utils.config import get_project_base_path
+
+LOGGER = logging.getLogger(__name__)
+
+
+def _sha256(s):
+    return None if pd.isna(s) else hashlib.sha256(s.encode("utf-8")).hexdigest()
+
+
+class DatasetAggregator:
+    def __init__(self, files: pd.DataFrame, config: dict):
+        self._config = config
+
+        self.files_in_scope = files.assign(url_hash=lambda df: df["url"].apply(_sha256))
+
+        self.data_folder = get_project_base_path() / (config["data_folder"])
+        self.data_folder.mkdir(parents=True, exist_ok=True)
+        self.combined_filename = get_project_base_path() / (config["combined_filename"])
+        self.errors = defaultdict(list)
+
+        self._add_filenames()
+
+    def run(self) -> None:
+        for file_infos in tqdm(self._files_to_run()):
+            if file_infos.format not in LOADER_CLASSES:
+                LOGGER.warning(f"Format {file_infos.format} not supported")
+                continue
+
+            if file_infos.url is None or pd.isna(file_infos.url):
+                LOGGER.warning(f"URL not specified for file {file_infos.title}")
+                continue
+
+            self._treat_file(file_infos)
+
+        self._concatenate_files()
+        with open(self.data_folder / "errors.json", "w") as f:
+            json.dump(self.errors, f)
+        return self
+
+    def _treat_file(self, file: tuple) -> None:
```

Comment:
nit: not sure sure treat is appropriate here
```suggestion
    def _process_file(self, file: tuple) -> None:
```